apps/analytics/views.py

```diff
@@ -107,15 +107,20 @@ def get_submission_count(request, challenge_pk, duration):
 @api_view(['GET', ])
 @permission_classes((permissions.IsAuthenticated, HasVerifiedEmail, IsChallengeCreator))
 @authentication_classes((ExpiringTokenAuthentication,))
-def get_challenge_phase_submission_analysis(request, challenge_pk, challenge_phase_pk):
+def get_challenge_phase_submission_analysis(request, challenge_pk, challenge_phase_id):
     """
     API to fetch
     1. The submissions count for challenge phase.
     2. The participated team count for challenge phase.
     """
     challenge = get_challenge_model(challenge_pk)
 
-    challenge_phase = get_challenge_phase_model(challenge_phase_pk)
+    challenge_phase = get_challenge_phase_from_phase_id(challenge, challenge_phase_id)
```

Comment:
@isht3 Please add this function in `apps/base/utils.py` instead of the above one.
`def get_model_object_by_id(model_name):
    """
    A function to get the model object by id in which challenge model acts as a foreign key.
    """

    def get_model_by_id(idx, challenge):
        try:
            model_object = model_name.objects.get(challenge=challenge, phase_id=idx)
            return model_object
        except model_name.DoesNotExist:
            raise NotFound('{} {} does not exist'.format(model_name.__name__, idx))
    get_model_by_id.__name__ = 'get_{}_object'.format(model_name.__name__.lower())
    return get_model_by_id`
Please test it on your local once, I think things will work fine with it.

Reply:
@RishabhJain2018 Every time I try to import Challenge models onto `base/utils.py` it gives me `ImportError: cannot import name ChallengePhase` when doing `from challenges.models import ChallengePhase`. Also would you please explain why we are using the above decorator instead of the other function?

Reply:
@isht3 I am not sure why you are getting this error as I checked and it is working for me. Can you please check once.
Moreover,  I am asking you to add this decorator so that we don't have to handle the error response separately everywhere.

Reply:
@RishabhJain2018 The CI tests are failing because of that. It's because of the import loop from having `base.utils` inside the `challenge.models`. Is it okay if I move this function to `challenge.utils` to avoid the import loops?